### ISSUE TITLE
Render leisure=resort

### ISSUE THREAD

**Issue by zetx16** [gravitystorm/openstreetmap-carto#2290]
Render leisure=resort and label.
Object for example: http://www.openstreetmap.org/way/290825122
I think the color is set like a tourism=camp_site (`@campsite: #def6c0;`)

**Comment by pnorman** [comment]
notes from taginfo

1.5k uses, usage is not worldwide
![image](https://cloud.githubusercontent.com/assets/1190866/17763494/f0cfe1c0-64cd-11e6-8814-005c57aed7c2.png)

**Comment by jojo4u** [comment]
The [wiki article](https://wiki.openstreetmap.org/wiki/Tag:leisure%3Dresort) created in 2015 is ok There was no proposal but some [discussion ](https://lists.openstreetmap.org/pipermail/tagging/2013-February/012883.html) in 2013.

**Comment by d1g** [comment]
[RU:Tag:leisure=resort provides more information](https://wiki.openstreetmap.org/wiki/RU:Tag:leisure%3Dresort) about additional sub classes. We would definitely tag these objects, as they were requested by users (pre Oct 2015) and still functional. The only question is uniform  tagging.

Most of them have nothing to do with hotels/motels/hostels, in sense you wouldn't visit leisure=resort + resort=kids_camp as adult, unless your children are there.

You wouldn't visit other leisure objects unless your are ill/old/ and so on.

re @pnorman  I don't think that mostly outdoor facilities dedicated to children resorts leisure=resort + resort=kids_camp ("детский лагерь") can have global coverage. Not many countries make such distinction, am I wrong?

Wikipedia says that they are present at least in some other countries: https://en.wikipedia.org/wiki/Summer_camp#Around_the_world but are they used? 

_"In Canada, summer camps are very popular. About 70% of Canadian camps tend to be affiliated with organizations, while the rest are private."_

**Comment by pnorman** [comment]
They're present around the world and used, but leisure=resort is mainly used in one area.

**Comment by kocio-pl** [comment]
They are concentrated around Europe, but they are also tagged all around the world, so it looks OK for me. I would render them just like other leisure/sport areas.

**Comment by polarbearing** [comment]
I'd see them more tourism-oriented, thus looking towards camp-sites as in the first post might be better than leisure/sport which is more everyday activities. 

We could also go a step forward and consider all campuses for touristic accommodation, hotel, resort, hostel etc, see #1624 (accommodation). 

I'd imagine a blue outline (similar to the brown for theme_parks)

**Comment by kocio-pl** [comment]
Makes sense for me.

**Comment by polarbearing** [comment]
Here are some experiments, using the code for theme_park in the blue from accommodation icons. 
Current (has recreation_ground and barrier):
![current](https://user-images.githubusercontent.com/8266355/35193754-af1f7720-fea7-11e7-962e-84094b85f220.png)
Blue outline, recreation_ground and barrier on:
![rec-ground_on](https://user-images.githubusercontent.com/8266355/35193760-cdc4bf96-fea7-11e7-9b5c-1d2408242712.png)
Blue outline, recreation_ground off, and barrier on:
![rec-ground-off](https://user-images.githubusercontent.com/8266355/35193764-dcc947c8-fea7-11e7-904b-bbcf0e2a4da4.png)
Blue outline, recreation_ground off, and barrier off:
![no-barrier](https://user-images.githubusercontent.com/8266355/35193766-ea41d1f4-fea7-11e7-8abd-d1c3ecbe5ffa.png)

I don't see a difference with barrier on or off, maybe the outline covers the barrier line?

Conflicts with maritime features need to be checked.

**Comment by kocio-pl** [comment]
It looks interesting, it's worth checking.

**Comment by polarbearing** [comment]
leisure=marina comes into mind, however it encloses a water body, and it is a different blue
![leisure-marina](https://user-images.githubusercontent.com/8266355/35199490-ca201884-fefe-11e7-8acd-15c4e5897062.png)
There seems to be no specific rendering of general harbours.

**Comment by kocio-pl** [comment]
For me this is different enough.

**Comment by polarbearing** [comment]
Ok, so should we apply this to all campuses of
- leisure=resort|beach_resort (4k,3k)
- tourism=hotel|hostel|camp_site|caravan_site|motel|chalet|summer_camp|resort (278k,36k,88k,22k,36k,33k,7,493)
?

**Comment by kocio-pl** [comment]
I think yes in general.

> I don't see a difference with barrier on or off, maybe the outline covers the barrier line?

We should check it with zoo, probably this problem is there already.

**Comment by polarbearing** [comment]
Code seems fine, I made some code tests with thick red fences. However the red blends with the tourism or accommodation colour. This is the same if I put the outline code before or after the barrier code, despite only the outline has opacity? Shouldn't the solid red paint clearly over, or would that have to be in a different file?
Anyway, so the normal 0.4 grey fences are probably just too weak in the outline to be visible. 
![red-fence-over-tourism-boundary](https://user-images.githubusercontent.com/8266355/35253645-e0d34d56-ffe6-11e7-8543-1e2e38e62b95.png)![red-fence-over-accommodation-boundary](https://user-images.githubusercontent.com/8266355/35253648-e5f6caf6-ffe6-11e7-9a82-dce35f2b5bd3.png)

**Comment by polarbearing** [comment]
I learnt that in the earlier tests, barrier was not really off, since the fence being a closed line, when the object also has an area (landuse?) tag, becomes an area_barrier. 

Since the outline code uses opacity, we get a few blending effects so that needs to be tested with different landuse backgrounds.

**Comment by kocio-pl** [comment]
I'm not sure if we need two lines in the first place. I guess they are good for really big areas, where you want to know what is inside and what is outside (like natural parks), but accommodation areas should be much smaller and light blue (even thinner than currently) could be enough.

**Comment by polarbearing** [comment]
Only one line, width=2, no opacity (avoids problems with changing landuse like commercial etc):
z16
![z16-2px](https://user-images.githubusercontent.com/8266355/35310745-354ca978-00b3-11e8-81c0-3f54ed2c74af.png)
z18
![z18-2px](https://user-images.githubusercontent.com/8266355/35310752-3d44ebae-00b3-11e8-9121-1bdc02ef98d4.png)

**Comment by polarbearing** [comment]
And brown for museums: (z18 width=2 amenity-brown)
![z18-museum-2px-brown](https://user-images.githubusercontent.com/8266355/35311670-fb07ac9a-00b7-11e8-9ac5-8c65d2761c95.png)

**Comment by kocio-pl** [comment]
Both are looking very nice for me. Things to check and decide:
1. What about the barrier visibility?
2. How do they compare to the river/stream and track (with tracktype=grade1) respectively?

**Comment by kocio-pl** [comment]
I think `tourism=resort` shouldn't be included, because both `leisure=resort` and `leisure=beach_resort` has more uses and are advised instead on the wiki.

**Comment by polarbearing** [comment]
Flooding the roads as streams and rivers:
z15
![z15_stream](https://user-images.githubusercontent.com/8266355/35353186-a5e3b634-0146-11e8-8b68-3eaf8587c6da.png)![z15_rivers](https://user-images.githubusercontent.com/8266355/35353193-ac71a196-0146-11e8-8284-0e1c923a2da1.png)
z17
![z17_stream](https://user-images.githubusercontent.com/8266355/35353205-b8bfc00e-0146-11e8-931b-66cd301ceae7.png)![z17_rivers](https://user-images.githubusercontent.com/8266355/35353212-bf7b0d2c-0146-11e8-89d2-156b696e85c5.png)
z19
![z19_stream](https://user-images.githubusercontent.com/8266355/35353223-ca4526de-0146-11e8-8ba0-b1c1047ca2d0.png)
![z19_rivers](https://user-images.githubusercontent.com/8266355/35353230-d1f028ca-0146-11e8-983b-c4e40ba6ede4.png)

**Comment by polarbearing** [comment]
Barrier tests, z19
blue line width=2 offset =-1
![z19_without_barrier](https://user-images.githubusercontent.com/8266355/35354053-4cee460e-0149-11e8-94fc-bdcadcbc600e.png)
![z19_with_barrier](https://user-images.githubusercontent.com/8266355/35354062-56727cb8-0149-11e8-9fac-19cce3409ec9.png)

offset=-1.5 improves barrier, but without you see 0.5 width of the landuse behind when no barrier is tagged.
![z19_without_barrier_offset_-1 5](https://user-images.githubusercontent.com/8266355/35354274-d377d88e-0149-11e8-9f39-43c4cb603ad2.png)

![z19_with_barrier_offset_-1 5](https://user-images.githubusercontent.com/8266355/35354108-6c32a1b8-0149-11e8-9495-0998b1605e1d.png)

Making the offset conditional on the existence of a barrier might be tricky since only parts of the outline might have a fence.

**Comment by polarbearing** [comment]
Trying a lighter colour for the outline (as in museum tests).
z18, outline colour equals icon colour:
![z18_transport-icon-color_offset-1 5](https://user-images.githubusercontent.com/8266355/35386389-2790f9d2-01cc-11e8-9bf0-fbb619d78a47.png)

z18, outline colour 17% lighter than icon, with barrier:
![z18_transport-lightercolor_offset-1_barrier](https://user-images.githubusercontent.com/8266355/35386423-494a8c6e-01cc-11e8-9e90-7e2067a3721c.png)

z18, outline colour 17% lighter than icon, without barrier:
![z18_transport-lightercolor_offset-1_barrier_off](https://user-images.githubusercontent.com/8266355/35386431-542df0e4-01cc-11e8-878e-4df7a1fd07c7.png)

**Comment by polarbearing** [comment]
So far, barriers were rendered before the tourism outlines. The latter need to be moved up in the project.mml so barriers draw over them. 
This will improve the current theme_park as well, ill show later.
![z19_transport-light_barrier_over_outline](https://user-images.githubusercontent.com/8266355/35392891-8c46487a-01e3-11e8-8100-19c3c13166c8.png)

**Comment by ghost07tula** [comment]
As I can see here the use of tag significantly increased worldwide last years. Just compare with one of the first comments on this issue. It seems the time has come to bring leisure=resort into focus.

**Comment by kocio-pl** [comment]
The main problem is to design how should it look like - if we have this, focus will follow most probably. :smile:

**Comment by polarbearing** [comment]
Probably I can come back to the outline tests the following weeks.

**Comment by EdLoach** [comment]
I suspect usage would be even higher if it rendered - I'm aware of a few places tagged tourism=resort or tourism=attraction where leisure=resort looks like it might be appropriate.

**Comment by jeisenbe** [comment]
If we pick a new color for accommodation icons, we could use a similar
color for the resort outline

**Comment by polarbearing** [comment]
Yes of course, that's the intention.

**Comment by jidanni** [comment]
https://www.openstreetmap.org/way/562255930 invisible in any layer.

**Comment by polarbearing** [comment]
Yes, that's why the ticket is still open.

**Comment by jidanni** [comment]
I'm wondering if the world's largest resorts show up in OSM...

Hmmm, looking at
https://www.touristmaker.com/blog/the-largest-resorts-in-the-world/
they mean resort towns.

https://www.cntraveler.com/gallery/best-resorts-in-the-world
are more like the resorts I'm thinking of.

Anyway let's first check if Club Med shows up...
https://www.openstreetmap.org/search?query=Club%20Med all not resorts...

**Comment by matkoniecz** [comment]
> Yes, that's why the ticket is still open.

Yes, in general commenting in opened issues (because issue was not yet fixed) that issue remains unfixed is not useful.

> openstreetmap.org/way/562255930 invisible in any layer.

Also, in this particular case it seems to be a mistagged feature anyway.

**Comment by jidanni** [comment]
> Also, in this particular case it seems to be a mistagged feature anyway.

OK I [retagged it](https://www.openstreetmap.org/changeset/68820167), based on their Facebook category. Thanks.

**Comment by PontiacCZ** [comment]
Current taginfo for [leisure=resort](https://taginfo.openstreetmap.org/tags/leisure=resort#overview):

Nearly 10K uses.

![image](https://user-images.githubusercontent.com/9153595/82831172-f7b10a00-9eb7-11ea-99ce-ccd4f43c1eda.png)

**Comment by jidanni** [comment]
I wonder if even [the most famous](https://www.google.com.tw/search?q=world%27s+most+famous+resorts) are on the map. Trump has one too in Florida...

**Comment by GunSmoker** [comment]
Can anyone suggest a possible workaround until this is implemented? leisure=resort + landuse=recreation_ground ? Anything at all?

**Comment by literannn** [comment]
13,7K uses... https://taginfo.openstreetmap.org/search?q=leisure%3Dresort

Maybe it's time to add rendering of leisure=resort to Carto?

**Comment by ghost07tula** [comment]
Yes, there are a lot of resorts in the database, but they are not rendered still.

**Comment by vazhnov** [comment]
Let's do it at least in 2022! Before the new summer season came — let's render all resorts…

**Comment by benoit-dd** [comment]
what's the consensus so far?
- the tag should be rendered (nobody opposed and it is widely used)
- label and icon color = accommodation blue

what's still to decide?
- fill and/or outline? what color?
- specific icon ? ==>usefull also for resort as node

Personnal comments:
I find the blue outline or filling disturbing as it is usally used by water element. 
Just an outline is fine as it doesn't interfere with landuse (for example some resort are cabins in the woods)
I'd go with the same line colour and style as theme_park, but it might be weird if the label+icon and line are not the same color?

For the icon, I'd suggest something with a few houses stacked together
![image](https://user-images.githubusercontent.com/52233014/156586664-5d385b33-f6e2-4059-8c0e-dc2a104d53d9.png)

**Comment by imagico** [comment]
This is an open discussion - anyone is welcome to present ideas and arguments.  But please understand that the presence of certain ideas in the discussion here (or the lack of critique of them) is not an indication of consensus.  A decision will only be made based on a concrete pull request.

I have not formed an opinion so far on this - nor do i have a clear understanding of what `leisure=resort` is de facto used for in OSM.  The wiki is decidedly non-helpful on that, it is vague and self contradicting.  Pages in different languages indicate that the tag might be used for semantically very different things in different parts of the world.

**Comment by benoit-dd** [comment]
I didn't intend to stop the discussion, but rather to ask id these two points where indeed consensual and see if it was possible to move to other points that where lacking to be able to make a satsifying pull-request

**Comment by BertMule** [comment]
I have added the tag to a building, then moved it to a point on the building.
Neither were rendered.
Purpose was to display the name.

Frustrating, as so often. 
Please do **something** with it.

**Comment by ZLima12** [comment]
Please don't map for the renderer.

**Comment by BertMule** [comment]
I map for the users.
No rendering = no use.

**Comment by jeisenbe** [comment]
@BertMule this issue is still open because no one has designed a suitable rendering or submitted a PR (pull request). 

All work on this map style is done by unpaid volunteers, and anyone is welcome to submit new rendering ideas and requests. 

See https://github.com/gravitystorm/openstreetmap-carto/blob/master/CONTRIBUTING.md for how to contribute a new PR, if you have the time.

**Comment by tommycrock** [comment]
I have a couple of suggestions for how this could be moved towards an acceptable PR and would appreciate feedback.
Although the meaning of resort is broad, it indicates accommodation and leisure facilities. The symbol could be the hotel/motel bed combined with the beach resort parasol. Not all resorts will have parasols but I think it's figurative enough.
Alternatively just render a name? These are often quite large areas that currently appear missing, which doesn't help mappers.

**Comment by harrija0** [comment]
I was just looking at resort areas and found them hard to distinguish without representation. I would love to see borders of resorts, all inclusive clubs, etc. rendered on the map. I see this as a safety and awareness thing as much as a visual appeal.

**Comment by JasonRaveling** [comment]
It looks like the tag may be considered too broad, according to the conversation on pull request #5212. We may need to consider some additional tagging that is more specific.

**Comment by aceman444** [comment]
There is the 'resort' tag (https://wiki.openstreetmap.org/wiki/Key:resort) to more specifically describe the resort type.

**Comment by imagico** [comment]
Use of `resort=*` has been discussed a bit in https://github.com/openstreetmap-carto/openstreetmap-carto/pull/5212#issuecomment-4275523402.
